http/relay/relay.go

```diff
@@ -160,29 +182,40 @@ func (p *Proxy) environment() []string {
 
 // start the server on a free port.
 func (p *Proxy) start() error {
+	cmd := p.cmd
+	p.cmd = nil
+
+	// Send this previous command to be cleaned up (Waited on, killed if necessary)
+	p.cmdCleanup <- cmd
+
 	port, err := freeport.Get()
 	if err != nil {
 		return errors.Wrap(err, "getting free port")
 	}
-	p.port = port
+
 	ctx.Infof("found free port %d", port)
+	target, err := url.Parse(fmt.Sprintf("http://127.0.0.1:%d", port))
+	if err != nil {
+		return errors.Wrap(err, "parsing url")
+	}
+
+	p.port = port
+	p.target = target
```

Comment:
I was trying to keep the property that if we return an error, `p` is not mutated.

Reply:
SGTM! I'll try and clean up my junk a bit soonish

Reply:
tightened this up even more.

Reply:
just kidding, this caused a pretty bad bug - reverting.